patches/tModLoader/ReLogic/Content/Sources/FileSystemContentSource.cs.patch

```diff
@@ -19,13 +23,18 @@
  
  	public int FileCount => _nameToAbsolutePath.Count;
  
-@@ -21,8 +_,12 @@
+@@ -21,8 +_,17 @@
  			_basePath += Path.DirectorySeparatorChar;
  
  		BuildNameToAbsolutePathDictionary();
 +
 +		// Added by TML.
 +		SetAssetNames(_nameToAbsolutePath.Keys);
++
++		// #2617 - Allow using paths with extensions in GetExtension calls, because we happen to return such paths in EnumerateAssets().
++		foreach (var pair in assetExtensions.ToArray()) {
++			assetExtensions.TryAdd(pair.Key + pair.Value, pair.Value);
++		}
```

Comment:
Shouldn't this be in `SetAssetNames`? There are other content sources than just `FileSystemContentSource`

Reply:
FSCS is the only content source that returns paths with extensions in EnumerateAssets. I went for the most minimal solution with the least chance of breaking anything further, as this fix is intended to be cherrypicked to stable. Future asset system reworks from #1612 can make more permanent improvements.